pytensor/tensor/subtensor.py

```diff
@@ -2686,6 +2687,32 @@ def grad(self, inputs, grads):
 advanced_subtensor = AdvancedSubtensor()
 
 
+@_vectorize_node.register(AdvancedSubtensor)
+def vectorize_advanced_subtensor(op: AdvancedSubtensor, node, *batch_inputs):
+    x, *idxs = node.inputs
+    batch_x, *batch_idxs = batch_inputs
+
+    # If indexes are batched fallback to Blockwise
+    if any(
+        batch_idx.type.ndim > idx.type.ndim
+        for batch_idx, idx in zip(batch_idxs, idxs)
+        if isinstance(batch_idx, TensorVariable)
```

Comment:
Is this condition necessary? When would a `batch_idx` not be a `TensorVariable`?

Reply:
They can be variables of SliceType or NoneType